mlx/random.cpp

```diff
@@ -192,6 +193,62 @@ array normal(
   return samples;
 }
 
+array multivariate_normal(
+    const array& mean,
+    const array& cov,
+    const std::vector<int>& shape,
+    Dtype dtype,
+    const std::optional<array>& key,
+    StreamOrDevice s) {
+  auto stream = to_stream(s);
+
+  if (dtype != float32) {
+    throw std::invalid_argument("[multivariate_normal] dtype must be float32.");
+  }
+
+  auto n = *(mean.shape().end() - 1);
+
+  // Check shapes comatibility of mean and cov
+  if (n != *(cov.shape().end() - 2)) {
+    throw std::invalid_argument(
+        "[multivariate_normal] last two dimensions of cov must be equal.");
+  }
+  if (*(mean.shape().end() - 1) != *(cov.shape().end() - 1)) {
+    throw std::invalid_argument(
+        "[multivariate_normal] mean and cov must have compatible shapes.");
+  }
+
+  // Compute output shape
+  std::vector<int> truncated_output_shape;
+  try {
+    auto truncated_mean_shape =
+        std::vector<int>(mean.shape().begin(), mean.shape().end() - 1);
+    auto truncated_cov_shape =
+        std::vector<int>(cov.shape().begin(), cov.shape().end() - 2);
+    auto trucated_inferred_shape =
+        broadcast_shapes(truncated_cov_shape, truncated_cov_shape);
+    truncated_output_shape = broadcast_shapes(trucated_inferred_shape, shape);
+  } catch (const std::invalid_argument& e) {
+    throw std::invalid_argument(
+        "[multivariate_normal] mean, cov and shape must have compatible shapes.");
+  }
+  std::vector<int> output_shape = {n};
+  output_shape.insert(
+      output_shape.begin(),
+      truncated_output_shape.begin(),
+      truncated_output_shape.end());
```

Comment:
I don't think you need a new shape vector here. Just push `n` back on the `truncated_output_shape` (and maybe rename it to just `output_shape`.